docs/source/python/memory.rst

```diff
@@ -277,6 +277,95 @@ types than with normal Python file objects.
    !rm example.dat
    !rm example2.dat
 
+Efficiently Writing and Reading Arrow Arrays
+--------------------------------------------
+
+Being optimized for zero copy and memory mapped data, Arrow allows to easily
+read and write arrays consuming the minimum amount of resident memory.
+
+When writing and reading raw arrow data, we can use the Arrow File Format
+or the Arrow Streaming Format.
+
+To dump an array to file, you can use the :meth:`~pyarrow.ipc.new_file`
+which will provide a new :class:`~pyarrow.ipc.RecordBatchFileWriter` instance
+that can be used to write batches of data to that file.
+
+For example to write an array of 100M integers, we could write it in 1000 chunks
+of 100000 entries:
+
+.. ipython:: python
```

Comment:
I know we've started using those `ipython` blocks, but I'm really not fond of them. They seem to make building docs quite slower (especially if the workload is non-trivial).

@jorisvandenbossche What do you think?

Reply:
I don't have a strong opinione about using `ipython` blocks or just code blocks. I used them just for consistency with the rest of the document.

I think that they have a value in at least verifying that the code you provided as an example can actually execute (even thought it might lead to different results) which makes more easy to catch examples that became invalid due to api changes.

Reply:
In this case in particular, we're creating and serializing a large amount of data, so that would really seem to add some cost to building the docs.

Reply:
Personally I would prefer to have *some* way to still verify the example, but this doesn't need to be with the IPython directive (which actually only ensures the code runs without error, not that the output is correct). This has come up before as well, so I opened a separate JIRA to discuss this in general: https://issues.apache.org/jira/browse/ARROW-13159

Reply:
For this specific example, we could also make it smaller? (which is still OK for educational purposes I think?)

Reply:
Reduced the size of the example to 1/10th